plugins/CuraEngineBackend/CuraEngineBackend.py

```diff
@@ -148,31 +154,46 @@ def getEngineCommand(self):
     ##  Emitted when the slicing process is aborted forcefully.
     slicingCancelled = Signal()
 
+    @pyqtSlot()
+    def stopSlicing(self):
+        self.backendStateChange.emit(BackendState.NotStarted)
+        if self._slicing:  # We were already slicing. Stop the old job.
+            self._terminate()
+            self._createSocket()
+
+        if self._process_layers_job:  # We were processing layers. Stop that, the layers are going to change soon.
+            self._process_layers_job.abort()
+            self._process_layers_job = None
+
+        if self._error_message:
+            self._error_message.hide()
+
+    ##  Manually triggers a reslice
+    @pyqtSlot()
+    def forceSlice(self):
+        if self._use_timer:
+            self._change_timer.start()
+        else:
+            self.slice()
+
     ##  Perform a slice of the scene.
     def slice(self):
-        Logger.log("d", "Starting slice job...")
-        if self._pause_slicing or self._block_slicing:
-            return
         self._slice_start_time = time()
-        if not self._enabled or not self._global_container_stack:  # We shouldn't be slicing.
-            # try again in a short time
-            self._change_timer.start()
+        if not self._need_slicing:
+            self.processingProgress.emit(1.0)
+            self.backendStateChange.emit(BackendState.Done)
+            Logger.log("w", "Do not need to slice.")
```

Comment:
I find this a vague error message. Did not need to slice or "Nothing to slice" is better imho. (although I don't feel very strong about this)